examples/flattened_enum.rs

```diff
@@ -0,0 +1,115 @@
+//! This example demonstrates how to deserialize enum nodes using an intermediate
+//! custom deserializer.
+//! The `elem` node can either be a `Foo` or a `Bar` node, depending on the `type`.
+//! The `type` attribute is used to determine which variant to deserialize.
+//! This is a workaround for [serde's issue](https://github.com/serde-rs/serde/issues/1905)
+//!
+//! note: to use serde, the feature needs to be enabled
+//! run example with:
+//!    cargo run --example flattened_enum --features="serialize"
+
+use std::fmt;
+
+use quick_xml::de::from_str;
+use serde::Deserialize;
+use serde::{
+    de::{self, value::MapAccessDeserializer, MapAccess, Visitor},
+    Deserialize,
+};
```

Comment:
This line 14 is redundant because you add this import in line 17
Although I prefer to not use nested imports.

```suggestion
use serde::{Deserialize, Deserializer};
use serde::de::{Error, MapAccess, Visitor};
use serde::de::value::MapAccessDeserializer;
```

and replace `serde::Deserializer` with `Deserializer` and `de::Error` with `Error`

Reply:
Fixed; thanks!